core/server/api/index.js

```diff
@@ -265,6 +213,15 @@ http = function http(apiMethod) {
             options = {};
         }
 
+        if (req.method === 'POST') {
```

Comment:
I would suggest to move this logic after calling the api method.
This more accurate.